actions/docxUpdater.js

```diff
@@ -0,0 +1,141 @@
+* property laws, including trade secret and copyright laws.
+* Dissemination of this information or reproduction of this material
+* is strictly forbidden unless prior written permission is obtained
+* from Adobe.
+************************************************************************* */
+const parseMarkdown = require('milo-parse-markdown').default;
+const { mdast2docx } = require('../node_modules/milo-md2docx/lib/index');
+const { getAioLogger } = require('./utils');
+const { fetchWithRetry } = require('./sharepoint');
+
+const gbStyleExpression = 'gb-';//graybox style expression. need to revisit if there are any more styles to be considered.
+const emptyString = '';
+const grayboxStylesRegex = new RegExp('gb-[a-zA-Z0-9,._-]*', 'g');
+const gbDomainSuffix = '-graybox';
+const logger = getAioLogger();
+let firstGtRows = [];
+
+
+/**
+ * Updates a document based on the provided Markdown file path, experience name, and options.
+ * @param {string} mdPath - The path to the Markdown file.
+ * @param {string} experienceName - The name of the experience.
+ * @param {object} options - The options for fetching the Markdown file.
+ * @returns {Promise} - A promise that resolves to the generated Docx file.
+ */
+async function updateDocument(mdPath, expName, options = {}){
+    firstGtRows = [];
+    const response = await fetchWithRetry(`${mdPath}`, options);
+    const content = await response.text();
+    if (content.includes(expName) || content.includes(gbStyleExpression) || content.includes(gbDomainSuffix)) {
+        const state = { content: { data: content }, log: '' };
+        await parseMarkdown(state);
+        const { mdast } = state.content;
+        updateExperienceNameFromLinks(mdast.children, expName);
+        console.log('All links replaced');
+        iterateGtRowsToReplaceStyles();
+        console.log('All styles replaced');
+        //generated docx file from updated mdast
+        const docx = await generateDocxFromMdast(mdast);
+        //TODO promote this docx file
+        console.log('Mdast to Docx file conversion done');
+    }
+}
+
+/**
+ * Replace all relative link references in the given mdast with the provided experience name and graybox style pattern.
+ * @param {Array} mdast - The mdast to be updated.
+ * @param {string} expName - The name of the experience.
+ * @param {RegExp} grayboxStylePattern - The pattern to match graybox styles.
+ */
+const updateExperienceNameFromLinks = (mdast, expName) => {
+    if (mdast) {
+        mdast.forEach((child) => {
+                if (child.type === 'gridTable') {
+                    firstGtRows.push(findFirstGtRowInNode(child));
+                }
+                //remove experience name from links on the document
+                if (child.type === 'link' && child.url && (child.url.includes(expName) || child.url.includes(gbDomainSuffix))) {
+                    child.url = child.url.replaceAll(expName, emptyString).replaceAll(gbDomainSuffix, emptyString);
```

Comment:
`child.url.replaceAll(expName, emptyString)`
If `expName` has just the name, will this not leave double slash in the link? Perhaps it should replace `/expName/` with `/` ?